File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    deliveries=deliveries[::-1]
    pickups=pickups[::-1]
    
    cnt_deliver=0
    cnt_pickup=0
    
    answer=0
    for i in range(n):
        cnt_deliver+=deliveries[i]
        cnt_pickup+=pickups[i]
        
        while cnt_deliver>0 or cnt_pickup>0:
            cnt_deliver -= cap
            cnt_pickup -= cap
            
            answer+= (n-i)*2
    
    return answer

File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.java
import java.util.*;

class Solution {
    public long solution(int cap, int n, int[] deliveries, int[] pickups) {
        long answer = 0;
        int len = deliveries.length;
        
        int cntDelivery = 0;
        int cntPickups=0;
        for(int i= len-1; i>=0; i--){
            cntDelivery += deliveries[i];
            cntPickups += pickups[i];
            
            while(cntDelivery > 0 || cntPickups > 0){
                cntDelivery -= cap;
                cntPickups -= cap;
                
                answer += (i+1)*2;
            }
        }
        return answer;
    }
}